addons/mail/models/mail_thread.py

```diff
@@ -1181,7 +1183,10 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
 
                 mail_message = self.env['mail.message']
                 if email_part:
-                    email = email_part.get_payload()[0]
+                    if email_part.is_multipart():
+                        email = email_part.get_payload()[0]
+                    else:
+                        email = Parser(policy=default).parsestr(email_part.get_payload())
```

Comment:
Out of consistency with how we do that in `message_parse()`, just use `message_from_string()`, it should work fine.
Even better: let's avoid shadowing the `email` package, and rename that temporary `email` variable to `bounced_eml` or something.

```suggestion
                        email = email.message_from_string(email_part.get_payload())
```